src/views/HomeView.tsx

```diff
@@ -1,9 +1,21 @@
 import HeaderBarWrapper from "@src/components/HeaderBarWrapper";
 import { NavBar } from "@src/components/NavBar";
 import { TopicCard } from "@src/components/PostCard";
-import { ScrollView, StyleSheet, Text, View } from "react-native";
+import { StyleSheet, Text, View, FlatList } from "react-native";
+import { getAllTopics } from "@src/services/viewService/getTopics";
+import { useQuery } from "@tanstack/react-query/build/lib/useQuery";
+import { returnTopicProps } from "@src/@types/returnProps";
 
 export const HomeView: React.FC = () => {
+  const { isLoading, isError, data, error } = useQuery({
+    queryKey: ["allTopics"],
+    queryFn: () => getAllTopics(),
+  });
```

Comment:
1. `queryFn` should be an async function.
2. `queryKey` is too simple, add some basic information
```suggestion
  const { data: topicList, isSuccess } = useQuery<ForumTopic[]>({
    queryKey: ["allTopics", "..."],
    queryFn: getAllTopics,
  });

  if (isSuccess) {
    return null; // or other view like a skeleton
  }

  // Now TS is convinced that data isn't undefined.
```.